packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -323,89 +457,84 @@ class _TileIcon extends StatelessWidget {
+       assert(showSelectedLabels != null),
+       assert(showUnselectedLabels != null),
+       super(key: key);
 
   final ColorTween colorTween;
   final Animation<double> animation;
   final BottomNavigationBarItem item;
+  final double selectedFontSize;
+  final double unselectedFontSize;
+  final bool showSelectedLabels;
+  final bool showUnselectedLabels;
 
   @override
   Widget build(BuildContext context) {
-    return Align(
-      alignment: Alignment.bottomCenter,
-      heightFactor: 1.0,
-      child: Container(
-        margin: const EdgeInsets.only(bottom: _kBottomMargin),
-        child: DefaultTextStyle.merge(
-          style: TextStyle(
-            fontSize: _kActiveFontSize,
-            color: colorTween.evaluate(animation),
-          ),
-          // The font size should grow here when active, but because of the way
-          // font rendering works, it doesn't grow smoothly if we just animate
-          // the font size, so we use a transform instead.
-          child: Transform(
-            transform: Matrix4.diagonal3(
-              Vector3.all(
-                Tween<double>(
-                  begin: _kInactiveFontSize / _kActiveFontSize,
-                  end: 1.0,
-                ).evaluate(animation),
-              ),
-            ),
-            alignment: Alignment.bottomCenter,
-            child: item.title,
+    Widget text = DefaultTextStyle.merge(
+      style: TextStyle(
+        fontSize: selectedFontSize,
+        color: colorTween.evaluate(animation),
+      ),
+      // The font size should grow here when active, but because of the way
+      // font rendering works, it doesn't grow smoothly if we just animate
+      // the font size, so we use a transform instead.
+      child: Transform(
+        transform: Matrix4.diagonal3(
+          Vector3.all(
+            Tween<double>(
+              begin: unselectedFontSize / selectedFontSize,
+              end: 1.0,
+            ).evaluate(animation),
           ),
         ),
+        alignment: Alignment.bottomCenter,
+        child: item.title,
       ),
     );
-  }
-}
-
-class _ShiftingLabel extends StatelessWidget {
-  const _ShiftingLabel({
-    Key key,
-    @required this.animation,
-    @required this.item,
-  }) : super(key: key);
 
-  final Animation<double> animation;
-  final BottomNavigationBarItem item;
+    if (!showUnselectedLabels && !showSelectedLabels) {
+      // Never show any labels.
+      text = Opacity(
+        alwaysIncludeSemantics: true,
+        opacity: 0.0,
+        child: text,
+      );
+    } else if (!showUnselectedLabels) {
+      // Fade selected labels in.
+      text = FadeTransition(
+        alwaysIncludeSemantics: true,
+        opacity: animation,
+        child: text,
+      );
+    } else if (!showSelectedLabels) {
+      // Fade selected labels out.
+      text = FadeTransition(
+        alwaysIncludeSemantics: true,
+        opacity: Tween<double>(begin: 1.0, end: 0.0).animate(animation),
```

Comment:
Couldn't you wrap this in a `ReverseAnimation` instead?

Reply:
I think so, however I find this (I didn't write this) easier to understand.

Reply:
Really? Because this seems a lot clearer to me (at least it would be to a newbie):
```suggestion
        opacity: ReverseAnimation(animation),
```

Reply:
The documentation for ReverseAnimation says:
```
Using a ReverseAnimation is different from simply using a Tween with a begin of 1.0 and an end of 0.0 because the tween does not change the status or direction of the animation.
```
So what are the implications of changing the animation's direction or status (he asked rhetorically).  While I'm scratching my head the person reading `Tween<double>(begin: 1.0, end: 0.0)` has already moved on.

Reply:
Well, but you could just as well scratch your head about what the .animate on the Tween does. Or the fact that a Tween isn't an animation, but you're assigning to one.

It's OK to keep it as a Tween, it's not a big deal, but I disagree that it's clearer.